### PR TITLE
Refreshed and improved to support bootstraping of major upgrade PR 🚀 

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Huge set of enhancements to the project, with main goals to refresh dependencies and to support the add hoc draft of major bump pull requests

- Refreshing :wind_face:
  - :traffic_light: test on latest node 522ce7cf446782a5f5946782443e6e9ec33f7a2b
  - :octopus: test on github actions 2701e91999d7fb4bb8624bcb90f9aab123b9fd1c
  - :outbox_tray: bump of major library, replacement of outdated ones (request, bluebird), modern rewrite of async code

- New features :rocket:
  - :tickets: ability to infer user & retrieve token from gh cli authentification 533384a9cec8d4ab55a25989f6f45464893d05ab
  - :open_file_folder: ability to run on a specific folder 90a347aa5d409e21c5662d4e7c57882f847d1792
  - :dart: ability to target a specific node version from command line b6329ca42e21f65abfd0de832a2f8ec956f615dc
  - :shield: force-with-lease by default (with optional override) d015ab0b6865f970605708576134da833c2e5f36
  - :memo: customizable reviewers and messages 361aeb4d46b10e9aac5cdf3566d8b4dbb4341a1c + 197b7b839701bfb453fcfd365a1d85a5e6e653a1
  - :fire_engine: preserve original engine range b557b6719539a921460258d5a06dd232c9193c8d

### CODE REVIEW

File: src/core/config.js [hunk 1]
```diff
@@ -57,12 +59,17 @@ const configSchema = Joi.object().keys({
 const resolver = protocall.getDefaultResolver();
 const readConfig = pathe => JSON.parse(fs.readFileSync(pathe));
 
+const resolveGithubToken = async argv => {
+  if (!argv.autoToken) return argv.token;
```
Comment (adamska27) [inline src/core/config.js:63]: you mean that, no?
```suggestion
  if (argv.autoToken) return argv.token;
```

File: src/core/git.js [hunk 2]
```diff
@@ -19,31 +19,30 @@ const commitFiles = async (branch, message) => {
...
+const headMessage = () => execSync('git', ['log', '-1', '--pretty=%B']).stdout;
+const headBranch = () => execSync('git', ['symbolic-ref', '--short', 'HEAD']).stdout;
+const currentUser = () => execSync('git', ['config', '--global', 'user.name']).stdout;
 
 const headClean = () => {
-  const res = childProcess
-    .execFileSync('git', ['status', '--porcelain'], {encoding: 'utf-8'})
-    .trim();
-  return res === '';
+  return execSync('git', ['status', '--porcelain']).stdout === '';
```
Comment (adamska27) [inline src/core/git.js:28]: --porcelain 
(I had to search to understand 😅 )

File: test/integration/package.json [hunk 3]
```diff
@@ -21,7 +21,8 @@
   "author": "Mr X @Coorp",
   "license": "MIT",
   "engines": {
-    "node": "^18.0.0"
+    "node": "^12.22.12",
```
Comment (adamska27) [inline test/integration/package.json]: why targeting node 12 ?

File: .eslintrc [hunk 4]
```diff
@@ -15,7 +15,8 @@
   ],
   "rules": {
     "no-continue": "off",
-    "fp/no-loops": "off"
+    "fp/no-loops": "off",
+    "no-nested-ternary": "off",
```
Comment (silouone) [inline .eslintrc:19]: 🙈

File: src/core/config.js [hunk 5]
```diff
@@ -74,9 +81,23 @@ const resolveConfig = async (config, configPath, argv) => {
     base.node.travis = defaultWithPath(base.node.travis, '.travis.yml');
     base.node.package = defaultWithPath(base.node.package, 'package.json');
   }
+  base.argv = argv;
+  base.forceFlag = argv.force ? '--force' : '--force-with-lease';
```
Comment (silouone) [inline src/core/config.js:85]: what are the scenario where you would use --force without lease  😂 ❓

File: src/core/config.js (hunk 1 above)
Comment (AdrieanKhisbe) [reply src/core/config.js:63]: Nope.

Logic is: if token is not to be inferred (auto token), return then eventual provided token

File: src/core/git.js (hunk 2 above)
Comment (AdrieanKhisbe) [reply src/core/git.js:28]: :amphora: for parseable format

File: test/integration/package.json (hunk 3 above)
Comment (AdrieanKhisbe) [reply test/integration/package.json]: mmmm, good catch. looks like an oversight!

File: src/core/config.js (hunk 5 above)
Comment (AdrieanKhisbe) [reply src/core/config.js:85]: I had to few time to use it doing testing.

File: test/integration/package.json (hunk 3 above)
Comment (AdrieanKhisbe) [reply test/integration/package.json]: @adamska27 the nvmrc wasn't updated in the node 18. So running code locally  restored the test version.
I just updated the all the version in the test/integration (in [bbfa790](https://github.com/CoorpAcademy/update-node/pull/143/commits/bbfa790fc79241645ff1f5c3fdb543be7bd73f22))

File: package.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
     "update-node": "./bin/update-node.js"
   },
   "engines": {
-    "node": "^18.17.1"
+    "node": ">=18.12.1"
```
Comment (StefanoA1) [inline package.json:10]: any reason why here it starts with the minor 12? I saw 18.14.2 below on other places

**Review by StefanoA1 — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (AdrieanKhisbe) [reply package.json:10]: Almost kind of random. 
- 18.12.1 is the oldest 18 LTS
- 18.14.2 was the oldest nvm installed version for node 18 on my device

**Review by esa-coorp — APPROVED** [review]

**Review by adamska27 — APPROVED** [review]

**Comment by AdrieanKhisbe** [comment]
:rocket: `@coorpacademy/update-node@7.0.0`

**Pull request merged**